From 24cf25f1919d0fef8da9e6ffd16d2bed0d3aa666 Mon Sep 17 00:00:00 2001
From: Allen Xu <b45815@freescale.com>
Date: Sat, 22 Nov 2014 13:15:19 -0600
Subject: [PATCH 0542/1074] MLK-9909 ARM: dts: add imx6sx sdb revA QSPI legacy
 support

add sdb revA board QSPI legacy support for Spansion QSPI chip.

Signed-off-by: Allen Xu <b45815@freescale.com>
(cherry picked from commit 3fd143ae78007e58d1dc45bc829b462a12b93cd2)
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sdb-reva.dts |   23 +++++++++++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb.dts      |    2 ++
 2 files changed, 25 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
index 36041f1..1107e34 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-reva.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
@@ -6,6 +6,7 @@
  * published by the Free Software Foundation.
  */
 
+#define SPANSIONFLASH
 #include "imx6sx-sdb.dts"
 
 /*
@@ -59,3 +60,25 @@
 		};
 	};
 };
+
+&qspi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_qspi2_1>;
+	status = "okay";
+
+	flash0: s25fl128s@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "spansion,s25fl128s";
+		spi-max-frequency = <66000000>;
+		reg = <0>;
+	};
+
+	flash1: s25fl128s@1 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "spansion,s25fl128s";
+		spi-max-frequency = <66000000>;
+		reg = <1>;
+	};
+};
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 79c1b70..623df2e 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -628,6 +628,7 @@
 	pinctrl-0 = <&pinctrl_qspi2_1>;
 	status = "okay";
 
+#ifndef SPANSIONFLASH
 	flash0: n25q256a@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -643,6 +644,7 @@
 		spi-max-frequency = <29000000>;
 		reg = <1>;
 	};
+#endif
 };
 
 &vadc {
-- 
1.7.5.4

